src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLQueryTypeImpl.java

```diff
@@ -584,6 +585,14 @@
 
   @Override
   public DataFetcher<RoutingResponse> plan() {
```

Comment:
We should rather change the output of the code generation to produce the interfaces to be typed AsyncDataFetcher<T>

Or we could automate this wrapping [here](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/ext/java/org/opentripplanner/ext/legacygraphqlapi/IntrospectionTypeWiring.java#L29) based on some condition. Possibly a static array of strings which have the fields we want to be async, and conditionally wrap the data fetcher.

Reply:
Both are really interesting suggestions. Let's discuss this in the meeting.